src/NeoSharp.Core/Blockchain/Processing/BlockProcessor.cs

```diff
@@ -54,19 +44,11 @@ public void Run()
             {
                 while (!cancellationToken.IsCancellationRequested)
                 {
-                    if (_blockchainContext.IsPeerConnected && _blockchainContext.NeedPeerSync && !_blockchainContext.IsSyncing)
-                    {
-                        _broadcaster.Broadcast(new GetBlocksMessage(_blockchainContext.CurrentBlock?.Hash));
-                        _blockchainContext.IsSyncing = true;
-                    }
+                    _blockPool.WaitForBlockBeenAddToPool(cancellationToken);
```

Comment:
This could be called for any block, we don't know for what block we are waiting for

Is this ok?

Reply:
yes .. that is why was removed.
this was added to get a way to get block in a certain order but was not working..

this piece of code should never been commit.